lib/msf/base/sessions/command_shell.rb

```diff
@@ -423,6 +426,40 @@ def repr(data)
     return data_repr
   end
 
+  def cmd_source_help
+    print_line("Usage: source [file] [background]")
+    print_line
+    print_line("Execute a local shell script file on remote machine")
+    print_line("This meta command will upload the script then execute it on the remote machine")
+    print_line
+    print_line("background")
+    print_line("`y` represent execute the script in background, `n` represent on foreground")
+  end
+
+  def cmd_source(*args)
+    if args.length != 2
+      # no argumnets, just print help message
+      return cmd_source_help
+    end
+
+    background = args[1].downcase == 'y'
+
+    local_file = args[0]
+    remote_file = "/tmp/." + ::Rex::Text.rand_text_alpha(32) + ".sh"
+
+    cmd_upload(local_file, remote_file)
+
+    if background
+      print_status("Executing on remote machine background")
+      print_line(shell_command("nohup sh -x #{remote_file} &"))
```

Comment:
Nice use of `xtrace` to log executed commands!

Reply:
Cool, I will try to test this command~ thank you for your suggestion~

Reply:
I mean the functionality of `set -x`, which is called `xtrace`:

```
wvu@kharak:~$ help set | grep -- -x | sed \$d
              xtrace       same as -x
      -x  Print commands and their arguments as they are executed.
wvu@kharak:~$
```